tests/unit/calculateShopBuyCost.test.ts

```diff
@@ -0,0 +1,38 @@
+import { describe, expect, test } from 'vitest';
+
+import calculateShopBuyCost from '../../src/lib/util/calculateShopBuyCost';
+
+describe('calculateShopBuyCost', () => {
+	test('fixed price cost', () => {
+		const result = calculateShopBuyCost(50, 3);
+		expect(result).toEqual({ total: 150, average: 50 });
+	});
+
+	test('variable cost within one world', () => {
+		const result = calculateShopBuyCost(100, 3, 1000, 10);
```

Comment:
Here is a good example of the previously recommended object params change for `calculateShopBuyCost`, looking at this, I have no idea what these numbers are.